### ISSUE TITLE
Multiple desktops being detected as one

### ISSUE THREAD

**Issue by yomono** [Ylianst/MeshCentral#2709]
Hi! 
I have this issue since the beginning of times, I just never reported it.
I can confirm this happens in a Ubuntu 20.10 desktop, with an Nvidia 1080ti.
Sadly I don't have a lot of dual monitor computers to try this haha.

When I connect, instead of getting two monitors recognized by the viewer, is just one very large monitor. You can see it here:

<details>
  <summary>Image</summary>
<img src="https://i.imgur.com/GjOudez.png">
</details>

I'm sure this is probably related to how Nvidia drivers manages the monitors, so I'm not sure if mesh central can do anything about it

**Comment by PetieM** [comment]
By default, you'll see all connected monitors in one view. Towards the bottom right, there should be a dropdown, at least when connecting to Windows devices, that lets you switch between them. It says "All Displays" by but you can select individual monitors from there as well.

**Comment by yomono** [comment]
Nope. That's exactly the issue, I have no option to switch in this case

**Comment by PetieM** [comment]
That's very odd. I don't have a Linux desktop with multiple displays to test with though if I had to guess, I'd say that's likely the problem (and arguably better than the current behavior on multi-monitor Macs which just ignores the other displays entirely). Display selection might be limited to Windows hosts at the moment, though don't quote me on that.

For reference, here's a screenshot of a connection being made to my own Windows-based PC, also with a 1080 Ti:

![image](https://user-images.githubusercontent.com/16453878/120123573-cfb05280-c16c-11eb-9a77-b58e37997051.png)

**Comment by Ylianst** [comment]
Bryan needs to look into this. I think it's been reported a few times that multi-monitor does not work on Linux and this needs to be fixed.

**Comment by krayon007** [comment]
On the console tab, can you run the following command: (ignore the cyclic input error)
`eval "require('monitor-info').getInfo(1000).then(function (m){sendConsoleText(JSON.stringify(m,null,1));});"`

**Comment by yomono** [comment]
> `eval "require('monitor-info').getInfo(1000).then(function (m){sendConsoleText(JSON.stringify(m,null,1));});"`

Hi! I tried this but I got this error:

>Command returned an exception error: TypeError: undefined not callable (property 'trim' of [object Function])

I just copied and pasted your command, is that ok?:

![image](https://user-images.githubusercontent.com/31445936/120357846-40c24800-c2dc-11eb-90eb-7308ebb066df.png)

Just in case, my user UID is 1000 endeed

**Comment by PetieM** [comment]
I just tried this on both a Windows PC and a Mac, both of which have multiple monitors connected.

Windows:
```
[
 {
  "left": 0,
  "top": 0,
  "right": 1024,
  "bottom": 768
 }
]
```

Mac:
```
Command returned an exception error: TypeError: undefined not callable (property 'getInfo' of [object Object])
```

**Comment by map7** [comment]
I'm also experiencing this issue on Linux computers and I have many with dual display. I'm running MeshCentral 1.0. I would love this feature to exist for Linux machines and happy to test.

**Comment by krayon007** [comment]
The crux of the issue, is that currently on linux, (from what I've seen on my systems at least), is that when I query the displays from the X server, it gives me a single virtual display that spans all the displays. I need to figure out how to get it to give me each display independently.

**Comment by map7** [comment]
On the command line of the linux client itself  the command 'xrandr' should give individual displays, maybe the meshagent isn't using this to pass back the individual displays to the meshcentral server.

**Comment by krayon007** [comment]
Using xrandr doesn't really help because I need to interact with the shared memory of the display object of the X server to do screen capture. So I need to be able to have the x server give me individual display objects.

**Comment by hojjatabdollahi** [comment]
@krayon007 Is using xcb an option?

I have a small sample code that reports the display servers, monitors for each one and then the shared memory for each screen.
It's not clean, but I can work on it if using xcb, xcb-randr, xcb-shm is an option. 

Or maybe this will give you some ideas as how to fix this issue for Linux?

[Here](https://gist.github.com/hojjatabdollahi/57323f9e2972329df54ac88b3a3a89de) is the snippet.

**Comment by pravardhanreddy** [comment]
Just checking to see if there are any updates on this. This would be an incredibly useful feature to have. I have about 10 users and all of us run Linux machines with multiple monitors. We would all greatly appreciate this feature.

**Comment by DovieW** [comment]
Unfortunate issue. Workaround is to mirror displays temporarily.

**Comment by yomono** [comment]
I don't have a lot of technical insights about this, but I'm almost sure the problem it's related to how Linux actually "reports" the monitors. The system is not actually reporting two monitors, it's just reporting one big screen with "co-ordinates", something like: from pixel x to 1920, this image will be display, and from 1921 to xxxx, this other image.
So it actually is just one big screen, so I guess the "fix" would need a complete re-implementation of this for Linux. Given the repo is barely maintained anymore and the fact that Linux represents a very small percentage of users, i don't have a lot of hopes.
Meshcentral still rocks, regardless of this!

**Comment by DaanSelen** [comment]
This issue is still present. Looks like MeshCentral sees it as one display. While Linux itself reports 2 displays:
`eval "require('monitor-info').getInfo(1000).then(function (m){sendConsoleText(JSON.stringify(m,null,1));});"`
```
[
 {
  "left": 0,
  "top": 0,
  "right": 3840,
  "bottom": 1080,
  "screen": {
   "_ObjectID": "_GenericMarshal.Variable",
   "_size": 8,
   "_LastError": 11
  },
  "screenId": 0,
  "display": {
   "_ObjectID": "_GenericMarshal.Variable",
   "_size": 8,
   "_LastError": 11
  }
 }
]
```

And Linux itself reports:
`xrandr`
```
Screen 0: minimum 320 x 200, current 3840 x 1080, maximum 16384 x 16384
VGA-1 disconnected (normal left inverted right x axis y axis)
DP-1 disconnected (normal left inverted right x axis y axis)
HDMI-1 connected 1920x1080+1920+0 (normal left inverted right x axis y axis) 476mm x 268mm
   1920x1080     60.00*+
   1680x1050     59.88  
   1600x900      60.00  
   1280x1024     60.02  
   1440x900      59.90  
   1280x720      60.00  
   1024x768      60.00  
   800x600       60.32  
   640x480       59.94  
DP-2 disconnected (normal left inverted right x axis y axis)
HDMI-2 connected primary 1920x1080+0+0 (normal left inverted right x axis y axis) 476mm x 268mm
   1920x1080     60.00*+
   1680x1050     59.88  
   1600x900      60.00  
   1280x1024     60.02  
   1440x900      59.90  
   1280x720      60.00  
   1024x768      60.00  
   800x600       60.32  
   640x480       59.94
```
